crates/ruma-client-api/src/dehydrated_device/delete_dehydrated_device.rs

```diff
@@ -0,0 +1,48 @@
+//! `DELETE /_matrix/client/*/dehydrated_device/`
+//!
+//! Delete a dehydrated device.
+
+pub mod unstable {
+    //! `msc3814` ([MSC])
+    //!
+    //! [MSC]: https://github.com/uhoreg/matrix-doc/blob/shrivelled_sessions/proposals/3814-dehydrated-devices-with-ssss.md
+
+    use ruma_common::{
+        api::{request, response, Metadata},
+        metadata, OwnedDeviceId,
+    };
+
+    const METADATA: Metadata = metadata! {
+        method: DELETE,
+        rate_limited: false,
+        authentication: AccessToken,
+        history: {
+            unstable => "/_matrix/client/unstable/org.matrix.msc3814.v1/dehydrated_device",
```

Comment:
Shouldn't the request include the device ID of the dehydrated device we want to delete? In case there's a race condition between two clients doing rehydration at the same time, and the other client already deleted the one we know about and replaced it with a new one?

Reply:
The intention here is that we can have only one dehydrated device. In the case of a race i think that we should™ be fine:

1. Device A is done and deletes the device while device B isn't done, device B can just abort and even upload a new dehydrated device.
2. Device A and device B are done at the same time, whichever newly uploaded dehydrated device comes later is fine, as long as one will get created before both devices, A and B, get deleted.